openmetadata-ui/src/main/resources/ui/src/axiosAPIs/searchAPI.ts

```diff
@@ -0,0 +1,196 @@
+} from '../interface/search.interface';
+import { omitDeep } from '../utils/APIUtils';
+import APIClient from './index';
+
+const getSearchIndexParam: (
+  si: SearchIndex | SearchIndex[] | undefined
+) => string | undefined = (si) => {
+  if (isNil(si)) {
+    return undefined;
+  }
+
+  if (isArray(si)) {
+    return si.join(',');
+  }
+
+  return si;
+};
+
+export const searchQuery = <
+  SI extends SearchIndex | SearchIndex[],
+  TIncludeFields extends KeysOfUnion<
+    SearchIndexSearchSourceMapping[SI extends Array<SearchIndex>
+      ? SI[number]
+      : SI]
+  >
+>(
+  req: SearchRequest<SI, TIncludeFields>
+): Promise<
+  SearchResponse<
+    SI extends Array<SearchIndex> ? SI[number] : SI,
+    TIncludeFields
+  >
+> => {
+  const {
+    query,
+    pageNumber = 1,
+    pageSize = 10,
+    queryFilter,
+    sortField,
+    sortOrder,
+    searchIndex,
+    includeDeleted,
+    trackTotalHits,
+    postFilter,
+    fetchSource,
+  } = req;
+
+  return (
+    APIClient.get<
+      SearchResponse<
+        SI extends Array<SearchIndex> ? SI[number] : SI,
+        TIncludeFields
+      >
+    >('/search/query', {
+      params: {
+        q: query,
+        index: getSearchIndexParam(searchIndex),
+        from: (pageNumber - 1) * pageSize,
+        size: pageSize,
+        deleted: includeDeleted,
+        /* eslint-disable @typescript-eslint/camelcase */
+        query_filter: JSON.stringify(queryFilter),
+        post_filter: JSON.stringify(postFilter),
+        sort_field: sortField,
+        sort_order: sortOrder,
+        track_total_hits: trackTotalHits,
+        fetch_source: fetchSource,
+        include_source_fields: req.fetchSource ? req.includeFields : undefined,
+        /* eslint-enable @typescript-eslint/camelcase */
+      },
+    })
+      .then(({ data }) => data)
+      // Elasticsearch responses use 'null' for missing values, we want undefined
+      // omitDeep is untyped, so this will return unknown, which is not necessarily bad
+      // since we need to do some more transformations to the response
+      .then((data) => omitDeep(data, isNil))
+      /* Elasticsearch objects use `entityType` to track their type, but the EntityReference interface uses `type`
+      This copies `entityType` into `type` (if `entityType` exists) so responses implement EntityReference */
+      .then((data) => ({
+        ...data,
+        hits: {
+          ...data.hits,
+          hits: isArray(data.hits.hits)
+            ? data.hits.hits.map((hit: { _source: Record<string, unknown> }) =>
+                '_source' in hit
+                  ? 'entityType' in hit._source
+                    ? {
+                        ...hit,
+                        _source: {
+                          ...hit._source,
+                          type: hit._source.entityType,
+                        },
+                      }
+                    : hit
+                  : hit
+              )
+            : [],
+        },
+      }))
```

Comment:
I am okay with having that util call within API as well but let's move it to util to keep API clean

Reply:
Moved to `async/await`.

I strongly think the processing should not be visible from outside the API, it's an implementation detail; components that use search shouldn't know about it.

@harshach @Sachin-chaurasiya what is your take on this?

Reply:
Hello @mosiac1, I agree processing is not doing much but it would be great if we move the processing part to util, because I think the API function should only make the API call and return the data, and processing should be done by the component that calls this API function using the util.

Also, there could be a case that the component wants to use this API function and want to process data in a different way.

Reply:
Fair enough will move it to a util function.

Reply:
Done, please re-review when you can @chirag-madlani